_targets.R

```diff
@@ -145,61 +142,82 @@ list(
 
   ),
 
-  ## summarize the data for each site parameter combination
+  # Link field notes to data stream, average observations if finer resolution
+  # than 15 minutes (this is rare, but it does happen sometimes).
   # KATIE REQUEST: HERE IS WHERE WE SHOULD JOIN/CBIND BATTERY AND BARO DATA,
   # AND REMOVE FROM FUTURE STEPS
+  # tar_target(
+  #   name = all_data_summary_list, # to do (j): name this something else
+  #   command = {
+  #     # field_notes
+  #     map2(.x = site_param_combos$sites,
+  #          .y = site_param_combos$params,
+  #          ~summarize_site_param(site_arg = .x,
+  #                                parameter_arg = .y,
+  #                                api_data = incoming_data_collated_csvs,
+  #                                notes = field_notes)) %>% # to do (j): why do I need to call field_notes here? I don't think we should need to do that...
+  #       set_names(paste0(site_param_combos$sites, "-", site_param_combos$params)) %>%
+  #       keep(~ !is.null(.))
+  #   },
+  #   packages = c("tidyverse", "padr")
+  # ),
   tar_target(
     name = all_data_summary_list, # to do (j): name this something else
-    command = {
-      # field_notes
-      map2(.x = site_param_combos$sites,
-           .y = site_param_combos$params,
-           ~summarize_site_param(site_arg = .x,
-                                 parameter_arg = .y,
-                                 api_data = incoming_data_collated_csvs,
-                                 notes = field_notes)) %>% # to do (j): why do I need to call field_notes here? I don't think we should need to do that...
-        set_names(paste0(site_param_combos$sites, "-", site_param_combos$params)) %>%
-        keep(~ !is.null(.))
-    },
+    command = summarize_site_param(site_arg = site_param_combos$sites,
+                                   parameter_arg = site_param_combos$params,
+                                   api_data = incoming_data_collated_csvs,
+                                   notes = field_notes) %>% # to do (j): why do I need to call field_notes here? I don't think we should need to do that...
+      set_names(paste0(site_param_combos$sites, "-", site_param_combos$params)) %>%
+      keep(~ !is.null(.)),
+    pattern = map(site_param_combo), # look up other pattern options, might be other things you can do here
```

Comment:
question: was this ran and compared to the old output?

the old command calls for a map2 over the columns in site_params_combos, but this uses just the one df. does `pattern = map(site_param_combo` perform this action row wise? I am a little confused how this works.